cgi-bin/help.c

```diff
@@ -173,7 +173,7 @@ main(int  argc,				/* I - Number of command-line arguments */
   if (cgiGetVariable("CLEAR"))
     cgiSetVariable("QUERY", "");
 
-  query = cgiGetTextfield("QUERY");
+  query = ( query = (char*)cgiGetTextfield("QUERY")) != NULL ? strdup(query) : NULL;
```

Comment:
Ok this code is hard to follow. Try refactoring, something like:

```c
const char *query = NULL; // …

…

if (cgiGetVariable(“CLEAR”))
  cgiSetVariable(“QUERY”, “”);
else if ((query = cgiGetTextField(“QUERY”)) != NULL)
  query = strdup(query);
```

Reply:
Thanks, updated!